### PR TITLE
Add new command to get active cluster.json configuration

### LINKED ISSUE #14983: Add command to obtain active cluster configuration
## Description

Sometimes it is necessary to obtain the active configuration of a cluster node. Right now there is no way to do it, the only alternative being to manually access the `cluster.json` file:
```
/var/ossec/framework/python/lib/python<PYTHON_VERSION>/site-packages/wazuh-<WAZUH_VERSION>-py<PYTHON_VERSION>.egg/wazuh/core/cluster/cluster.json
```

The problem is that any modification to that file does not take effect until the `clusterd` process is restarted. As a consequence, it is not always possible to know which is the active configuration of the cluster at a given moment.

Therefore, a new command must be implemented in the cluster that allows the API or any client to obtain the active configuration of the local node (the one in which the command is received).

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

### PR DESCRIPTION
**Pull request by Selutario**
|Related issue|
|---|
| Closes #14983 |

## Description

This PR adds a new cluster command, `get_cl_conf`, to obtain the active `cluster.json` configuration of the node where it is run. As this file is not intended to be modified by users, no API endpoint is required. A CLI option (`-rc`, `--read-config`) has been added to `/var/ossec/bin/cluster_control` instead:
```
# /var/ossec/bin/cluster_control   
usage: cluster_control.py [-h] [-d] [-fn [FILTER_NODE ...]] [-fs [FILTER_STATUS ...]] [-a | -l | -i [HEALTH] | -u | -rc]

optional arguments:
  -h, --help            show this help message and exit
  -d, --debug           Enable debug mode
  -fn [FILTER_NODE ...], --filter-node [FILTER_NODE ...]
                        Filter by node name
  -fs [FILTER_STATUS ...], --filter-agent-status [FILTER_STATUS ...]
                        Filter by agent status
  -a, --list-agents     List agents
  -l, --list-nodes      List nodes
  -i [HEALTH], --health [HEALTH]
                        Show cluster health
  -u, --usage           Show usage
  -rc, --read-config    Show active internal configuration
```

## Logs/Alerts example

```JSON
root@wazuh-master:/# /var/ossec/bin/cluster_control -rc 
{
    "files": {
        "etc/": {
            "permissions": 416,
            "source": "master",
            "files": [
                "client.keys"
            ],
            "recursive": false,
            "restart": false,
            "remove_subdirs_if_empty": false,
            "extra_valid": false,
            "description": "client keys file database"
        },
        "etc/shared/": {
            "permissions": 432,
            "source": "master",
            "files": [
                "all"
            ],
            "recursive": true,
            "restart": false,
            "remove_subdirs_if_empty": true,
            "extra_valid": false,
            "description": "shared configuration files"
        },
        "var/multigroups/": {
            "permissions": 432,
            "source": "master",
            "files": [
                "merged.mg"
            ],
            "recursive": true,
            "restart": false,
            "remove_subdirs_if_empty": true,
            "extra_valid": false,
            "description": "shared configuration files"
        },
        "etc/rules/": {
            "permissions": 432,
            "source": "master",
            "files": [
                "all"
            ],
            "recursive": true,
            "restart": true,
            "remove_subdirs_if_empty": false,
            "extra_valid": false,
            "description": "user rules"
        },
        "etc/decoders/": {
            "permissions": 432,
            "source": "master",
            "files": [
                "all"
            ],
            "recursive": true,
            "restart": true,
            "remove_subdirs_if_empty": false,
            "extra_valid": false,
            "description": "user decoders"
        },
        "etc/lists/": {
            "permissions": 432,
            "source": "master",
            "files": [
                "all"
            ],
            "recursive": true,
            "restart": true,
            "remove_subdirs_if_empty": false,
            "extra_valid": false,
            "description": "user CDB lists"
        },
        "excluded_files": [
            "ar.conf",
            "ossec.conf"
        ],
        "excluded_extensions": [
            "~",
            ".tmp",
            ".lock",
            ".swp"
        ]
    },
    "intervals": {
        "worker": {
            "sync_integrity": 9,
            "sync_agent_info": 10,
            "sync_agent_groups": 30,
            "timeout_agent_groups": 40,
            "keep_alive": 60,
            "connection_retry": 10,
            "max_failed_keepalive_attempts": 2,
            "agent_groups_mismatch_limit": 10
        },
        "master": {
            "process_pool_size": 2,
            "sync_agent_groups": 10,
            "timeout_extra_valid": 40,
            "recalculate_integrity": 8,
            "timeout_agent_info": 40,
            "agent_group_start_delay": 30,
            "check_worker_lastkeepalive": 60,
            "max_allowed_time_without_keepalive": 120,
            "max_locked_integrity_time": 1000
        },
        "communication": {
            "timeout_cluster_request": 20,
            "timeout_dapi_request": 200,
            "timeout_receiving_file": 120,
            "max_zip_size": 1073741824,
            "min_zip_size": 31457280,
            "compress_level": 1,
            "zip_limit_tolerance": 0.2
        }
    },
    "distributed_api": {
        "enabled": true
    }
}
```

In addition, the config is now logged to the `cluster.log` file every time the cluster process is started:
```
2023/03/15 09:26:24 INFO: [Cluster] [Main] Active configuration: {'files': {'etc/': {'permissions': 416, 'source': 'master', 'files': ['client.keys'], 'recursive': False, 'restart': False, 'remove_subdirs_if_empty': False, 'extra_valid': False, 'description': 'client keys file database'}, 'etc/shared/': {'permissions': 432, 'source': 'master', 'files': ['all'], 'recursive': True, 'restart': False, 'remove_subdirs_if_empty': True, 'extra_valid': False, 'description': 'shared configuration files'}, 'var/multigroups/': {'permissions': 432, 'source': 'master', 'files': ['merged.mg'], 'recursive': True, 'restart': False, 'remove_subdirs_if_empty': True, 'extra_valid': False, 'description': 'shared configuration files'}, 'etc/rules/': {'permissions': 432, 'source': 'master', 'files': ['all'], 'recursive': True, 'restart': True, 'remove_subdirs_if_empty': False, 'extra_valid': False, 'description': 'user rules'}, 'etc/decoders/': {'permissions': 432, 'source': 'master', 'files': ['all'], 'recursive': True, 'restart': True, 'remove_subdirs_if_empty': False, 'extra_valid': False, 'description': 'user decoders'}, 'etc/lists/': {'permissions': 432, 'source': 'master', 'files': ['all'], 'recursive': True, 'restart': True, 'remove_subdirs_if_empty': False, 'extra_valid': False, 'description': 'user CDB lists'}, 'excluded_files': ['ar.conf', 'ossec.conf'], 'excluded_extensions': ['~', '.tmp', '.lock', '.swp']}, 'intervals': {'worker': {'sync_integrity': 9, 'sync_agent_info': 10, 'sync_agent_groups': 30, 'timeout_agent_groups': 40, 'keep_alive': 60, 'connection_retry': 10, 'max_failed_keepalive_attempts': 2, 'agent_groups_mismatch_limit': 10}, 'master': {'process_pool_size': 2, 'sync_agent_groups': 10, 'timeout_extra_valid': 40, 'recalculate_integrity': 8, 'timeout_agent_info': 40, 'agent_group_start_delay': 30, 'check_worker_lastkeepalive': 60, 'max_allowed_time_without_keepalive': 120, 'max_locked_integrity_time': 1000}, 'communication': {'timeout_cluster_request': 20, 'timeout_dapi_request': 200, 'timeout_receiving_file': 120, 'max_zip_size': 1073741824, 'min_zip_size': 31457280, 'compress_level': 1, 'zip_limit_tolerance': 0.2}}, 'distributed_api': {'enabled': True}}
2023/03/15 09:26:24 INFO: [Local Server] [Main] Serving on /var/ossec/queue/cluster/c-internal.sock
2023/03/15 09:26:24 INFO: [Master] [Main] Serving on ('0.0.0.0', 1516)
```

## Tests

```
$ PYTHONPATH=/home/selu/Git/wazuh/api:/home/selu/Git/wazuh/framework python3 -m pytest framework -x --disable-warnings
============================================================================================ test session starts ============================================================================================
platform linux -- Python 3.9.16, pytest-7.0.1, pluggy-0.13.1
rootdir: /home/selu/Git/wazuh/framework
plugins: anyio-3.6.2, asyncio-0.18.1, aiohttp-1.0.4, cov-4.0.0
asyncio: mode=auto
collected 2099 items                                                                                                                                                                                        

framework/scripts/tests/test_agent_groups.py ..............                                                                                                                                           [  0%]
framework/scripts/tests/test_agent_upgrade.py ..............                                                                                                                                          [  1%]
framework/scripts/tests/test_cluster_control.py .......                                                                                                                                               [  1%]
framework/scripts/tests/test_rbac_control.py .........                                                                                                                                                [  2%]
framework/scripts/tests/test_wazuh_clusterd.py .......                                                                                                                                                [  2%]
framework/scripts/tests/test_wazuh_logtest.py ......................                                                                                                                                  [  3%]
framework/wazuh/core/cluster/dapi/tests/test_dapi.py ................................                                                                                                                 [  5%]
framework/wazuh/core/cluster/tests/test_client.py ................                                                                                                                                    [  5%]
framework/wazuh/core/cluster/tests/test_cluster.py ...................................                                                                                                                [  7%]
framework/wazuh/core/cluster/tests/test_common.py ..................................................................................                                                                  [ 11%]
framework/wazuh/core/cluster/tests/test_control.py .........                                                                                                                                          [ 11%]
framework/wazuh/core/cluster/tests/test_local_client.py ..............                                                                                                                                [ 12%]
framework/wazuh/core/cluster/tests/test_local_server.py .........................                                                                                                                     [ 13%]
framework/wazuh/core/cluster/tests/test_master.py ................................................                                                                                                    [ 15%]
framework/wazuh/core/cluster/tests/test_server.py .............................                                                                                                                       [ 17%]
framework/wazuh/core/cluster/tests/test_utils.py ...........                                                                                                                                          [ 17%]
framework/wazuh/core/cluster/tests/test_worker.py ..................................                                                                                                                  [ 19%]
framework/wazuh/core/tests/test_active_response.py ..............                                                                                                                                     [ 20%]
framework/wazuh/core/tests/test_agent.py ...................................................................................................................................................          [ 27%]
framework/wazuh/core/tests/test_cdb_list.py ......................................                                                                                                                    [ 28%]
framework/wazuh/core/tests/test_common.py .........                                                                                                                                                   [ 29%]
framework/wazuh/core/tests/test_configuration.py .......................................................................                                                                              [ 32%]
framework/wazuh/core/tests/test_database.py .............                                                                                                                                             [ 33%]
framework/wazuh/core/tests/test_decoder.py ................                                                                                                                                           [ 34%]
framework/wazuh/core/tests/test_exception.py ...                                                                                                                                                      [ 34%]
framework/wazuh/core/tests/test_input_validator.py ...                                                                                                                                                [ 34%]
framework/wazuh/core/tests/test_logtest.py ..                                                                                                                                                         [ 34%]
framework/wazuh/core/tests/test_manager.py ................                                                                                                                                           [ 35%]
framework/wazuh/core/tests/test_mitre.py .............                                                                                                                                                [ 35%]
framework/wazuh/core/tests/test_pyDaemonModule.py .....                                                                                                                                               [ 36%]
framework/wazuh/core/tests/test_results.py ........................................                                                                                                                   [ 38%]
framework/wazuh/core/tests/test_rootcheck.py .............                                                                                                                                            [ 38%]
framework/wazuh/core/tests/test_rule.py ......................                                                                                                                                        [ 39%]
framework/wazuh/core/tests/test_sca.py ........................                                                                                                                                       [ 40%]
framework/wazuh/core/tests/test_security.py .............                                                                                                                                             [ 41%]
framework/wazuh/core/tests/test_stats.py .................                                                                                                                                            [ 42%]
framework/wazuh/core/tests/test_syscheck.py .......                                                                                                                                                   [ 42%]
framework/wazuh/core/tests/test_syscollector.py ...                                                                                                                                                   [ 42%]
framework/wazuh/core/tests/test_task.py ........                                                                                                                                                      [ 43%]
framework/wazuh/core/tests/test_utils.py ............................................................................................................................................................ [ 50%]
......................................................................................                                                                                                                [ 54%]
framework/wazuh/core/tests/test_vulnerability.py ..                                                                                                                                                   [ 54%]
framework/wazuh/core/tests/test_wazuh_queue.py ....................                                                                                                                                   [ 55%]
framework/wazuh/core/tests/test_wazuh_socket.py ....................                                                                                                                                  [ 56%]
framework/wazuh/core/tests/test_wdb.py ...............................                                                                                                                                [ 58%]
framework/wazuh/core/tests/test_wlogging.py ............                                                                                                                                              [ 58%]
framework/wazuh/rbac/tests/test_auth_context.py ..                                                                                                                                                    [ 58%]
framework/wazuh/rbac/tests/test_decorators.py .............................................................................................................                                           [ 63%]
framework/wazuh/rbac/tests/test_default_configuration.py ........................................................                                                                                     [ 66%]
framework/wazuh/rbac/tests/test_orm.py .................................................................                                                                                              [ 69%]
framework/wazuh/rbac/tests/test_preprocessor.py ...........                                                                                                                                           [ 70%]
framework/wazuh/tests/test_active_response.py ............                                                                                                                                            [ 70%]
framework/wazuh/tests/test_agent.py ......................................................................................................................                                            [ 76%]
framework/wazuh/tests/test_cdb_list.py .....................................................                                                                                                          [ 78%]
framework/wazuh/tests/test_ciscat.py .................................                                                                                                                                [ 80%]
framework/wazuh/tests/test_cluster.py ..........                                                                                                                                                      [ 81%]
framework/wazuh/tests/test_decoder.py ...................................                                                                                                                             [ 82%]
framework/wazuh/tests/test_group.py .......                                                                                                                                                           [ 83%]
framework/wazuh/tests/test_logtest.py ......                                                                                                                                                          [ 83%]
framework/wazuh/tests/test_manager.py ....................................                                                                                                                            [ 85%]
framework/wazuh/tests/test_mitre.py .......                                                                                                                                                           [ 85%]
framework/wazuh/tests/test_rootcheck.py ..................................................                                                                                                            [ 87%]
framework/wazuh/tests/test_rule.py .......................................................                                                                                                            [ 90%]
framework/wazuh/tests/test_sca.py ...........                                                                                                                                                         [ 90%]
framework/wazuh/tests/test_security.py .......................................................................                                                                                        [ 94%]
framework/wazuh/tests/test_stats.py ...............                                                                                                                                                   [ 94%]
framework/wazuh/tests/test_syscheck.py .........................                                                                                                                                      [ 96%]
framework/wazuh/tests/test_syscollector.py ............                                                                                                                                               [ 96%]
framework/wazuh/tests/test_task.py ............................                                                                                                                                       [ 98%]
framework/wazuh/tests/test_vulnerability.py ........................................                                                                                                                  [100%]

=============================================================================== 2099 passed, 45 warnings in 302.12s (0:05:02) ===============================================================================
```

### CODE REVIEW

**Review by nico-stefani — DISMISSED** [review]
LGTM!

File: framework/scripts/cluster_control.py [hunk 1]
```diff
@@ -240,6 +248,7 @@ def main():
     exclusive.add_argument('-l', '--list-nodes', action='store_const', const='list_nodes', help='List nodes')
     exclusive.add_argument('-i', '--health', action='store', nargs='?', const='health', help='Show cluster health')
     exclusive.add_argument('-u', '--usage', action='store_true', help='Show usage')
+    exclusive.add_argument('-c', '--cluster-conf', action='store_true', help=argparse.SUPPRESS)
```
Comment (davidjiglesias) [inline framework/scripts/cluster_control.py:251]: -c could be a bit confusing honestly. I think something like rc (read config) makes more sense. C or c is sometimes used to provide config, not only reading it.

File: framework/wazuh/core/cluster/control.py [hunk 2]
```diff
@@ -196,10 +205,20 @@ async def get_node_ruleset_integrity(lc: local_client.LocalClient) -> dict:
...
+
+    Parameters
+    ----------
+    lc : LocalClient
+        LocalClient instance.
+
+    Returns
+    -------
+    dict
+        Dictionary with results
```
Comment (davidjiglesias) [inline framework/wazuh/core/cluster/control.py:222]:
```suggestion
        Dictionary with results.
```

**Review by davidjiglesias — CHANGES_REQUESTED** [review]
Also add the active configuration in the cluster.log at start, in INFO mode and in json format.

**Pull request closed without merging**